client/core/core.go

```diff
@@ -10419,9 +10431,18 @@ func (c *Core) SendShielded(appPW []byte, assetID uint32, toAddr string, amt uin
 		return nil, fmt.Errorf("password error: %w", err)
 	}
 
-	sw, err := c.shieldedWallet(assetID)
-	if err != nil {
-		return nil, err
+	w, found := c.wallet(assetID)
+	if !found {
+		return nil, fmt.Errorf("no %s wallet", unbip(assetID))
+	}
+
+	sw, is := w.Wallet.(asset.ShieldedWallet)
+	if !is {
+		return nil, fmt.Errorf("%s wallet is not a shielded wallet", unbip(assetID))
+	}
+
+	if !w.synchronized() {
```

Comment:
Would make sense for `ShieldFunds` and `UnshieldFunds`, possibly even `NewShieldedAddress` to check also, so maybe the `shieldedWallet` can return a bool too.

Reply:
`ShieldFunds` and `UnshieldFunds` are like balance checks, but I don't think they should be affected?

IMO, we could make it more obvious that wallets are still syncing and a tooltip about what they should not do until it's done. What do you think about this?

Reply:
> `ShieldFunds` and `UnshieldFunds` are like balance checks, but I don't think they should be affected?

No, these make transactions, sending value in transactions between accounts.  They should be blocked if the wallet is not synchronized.

Reply:
> No, these make transactions, sending value in transactions between accounts

Oh, let me check.

Reply:
Updated, but not checking in `NewShieldedAddress` yet unless you really think we should.